core/server/apps/amp/lib/router.js

```diff
@@ -42,21 +43,42 @@ function getPostData(req, res, next) {
             next();
         })
         .catch(function (err) {
-            if (err instanceof errors.NotFoundError) {
-                return next(err);
-            }
-
             next(err);
         });
 }
 
+function checkIfAMPIsEnabled(req, res, next) {
+    var ampIsEnabled = config.theme.amp;
+
+    if (ampIsEnabled) {
+        return next();
+    }
+
+    // CASE: we don't support amp pages for static pages
+    if (req.body.post && req.body.post.page) {
+        return errors.error404(req, res, next);
+    }
+
+    /**
+     * CASE: amp is disabled, we redirect the user to the requested post
```

Comment:
https://productforums.google.com/forum/#!topicsearchin/webmasters/amp$20redirect$20or$20404
https://shkspr.mobi/blog/2016/11/removing-your-site-from-amp/

People suggesting both, and people reporting 404 won't work well with SEO when they had AMP enabled for a while. I think it's a question of definition, because Ghost supports AMP and the user can enable and disable AMP. 

Let's use 404 and see if people complaining or not 👍 Will update quickly